### ISSUE TITLE
backprop(Layer, WorkspaceMgr) clears Dropout mask

### ISSUE THREAD

**Issue by dembanakh** [deeplearning4j/deeplearning4j#7641]
dl4j 1.0.0-beta3 as well as zoo

In previous issue, i point out a problem in StyleTransfer with Squeezenet.
Here i do the same thing (code from https://github.com/deeplearning4j/dl4j-examples/blob/master/dl4j-examples/src/main/java/org/deeplearning4j/examples/styletransfer/NeuralStyleTransfer.java), but now with DarkNet19 pretrained instead of VGG16.
The problem is I cannot execute StyleTransfer with more than 1 layer in STYLE_LAYERS array - after my looking into it, it seems that dropout masks are cleared after these line
`dLdAnext = layer.backpropGradient(dLdAnext, LayerWorkspaceMgr.noWorkspaces()).getSecond();` in backPropagate function

I get an error:
`Cannot perform backprop: Dropout mask array is absent (already cleared?)`
after the STYLE_LAYERS[1] gets backpropagated down beginning from its index. When it reaches STYLE_LAYERS[0] in graph, it says that there is no mask;

So, is it a bug or I do not pointed to the net the fact that i do not want to clear mask?

**Comment by dembanakh** [comment]
UPD: After I added Graph.feedForward() before every STYLE_LAYER it works. Thus, it means, that dropout masks are really got cleared after one backpropagation. Is there an option to forbid that or is it a bug?

**Comment by dembanakh** [comment]
I resolved the problem by storing Dropout masks for each layer in a Map.
